pwiz_tools/Shared/PanoramaClient/PanoramaUtil.cs

```diff
@@ -994,33 +1018,40 @@ public PanoramaServer ChangeUri(Uri uri)
             return ChangeProp(ImClone(this), im => im.URI = uri);
         }
 
-        public bool HasUserCredentials()
+        public bool HasUserAccount()
         {
-            return Username != null;
+            return Username?.Trim().Length > 0 && Password?.Trim().Length > 0;
         }
 
-        public bool RemoveContextPath()
+        public bool RemoveContextPath(out PanoramaServer panoramaServer)

```

Comment:
Immutable implementation should return a PanoramaServer, which can be "this" if the operation won't change anything.